UnityProject/Assets/Scripts/Items/Tool/TRayScanner.cs

```diff
@@ -0,0 +1,211 @@
+			if(info.ToSlot == null) return;
+
+			//To new player
+			if (info.ToPlayer != null)
+			{
+				//Put in backpack
+				if (info.ToSlot.Player == null)
+				{
+					//Should already be off so dont need to do anything
+					return;
+				}
+
+				//Ping to turn on if needed
+				DoRpc(info.ToPlayer, currentMode != Mode.Off);
+			}
+		}
+
+		private void DoRpc(RegisterPlayer player, bool newState)
+		{
+			if(player.connectionToClient == null) return;
+
+			if(CustomNetworkManager.IsServer && CustomNetworkManager.IsHeadless == false)
+			{
+				//Target RPC not working on local host?
+				if (newState)
+				{
+					DoState(currentMode);
+					return;
+				}
+
+				DoState(Mode.Off);
+				return;
+			}
+
+			RpcChangeState(player.connectionToClient, newState);
+		}
+
+		[TargetRpc]
+		private void RpcChangeState(NetworkConnection conn, bool newState)
+		{
+			if (newState)
+			{
+				DoState(currentMode);
+				return;
+			}
+
+			DoState(Mode.Off);
+		}
+
+		private void SyncMode(Mode oldMode, Mode newMode)
+		{
+			currentMode = newMode;
+			if(PlayerManager.LocalPlayerScript == null) return;
+
+			if(pickupable.ItemSlot == null || pickupable.ItemSlot.Player == null ||
+			   pickupable.ItemSlot.Player != PlayerManager.LocalPlayerScript.registerTile) return;
+
+			DoState(currentMode);
+		}
+
+		private void DoState(Mode newMode)
+		{
+			var matrixInfos = MatrixManager.Instance.ActiveMatricesList;
+
+			//TODO update for layer PR
+
+			foreach (var matrixInfo in matrixInfos)
+			{
+				var tilemapRenderer = matrixInfo.Matrix.UnderFloorLayer.GetComponent<TilemapRenderer>();
+				tilemapRenderer.sortingLayerName = newMode == Mode.Off ? "UnderFloor" : "Walls";
+				tilemapRenderer.sortingOrder = newMode == Mode.Off ? 0 : 1;
+			}
+		}
+
+		public void OnPlayerRejoin()
+		{
+			if(pickupable.ItemSlot?.Player == null) return;
+
+			DoRpc(pickupable.ItemSlot.Player, currentMode != Mode.Off);
+		}
+
+		public void OnPlayerTransfer()
+		{
+			if(pickupable.ItemSlot?.Player == null) return;
+
+			DoRpc(pickupable.ItemSlot.Player, currentMode != Mode.Off);
+		}
+
+		public void OnPlayerLeaveBody()
+		{
+			if(pickupable.ItemSlot?.Player == null) return;
+
+			DoRpc(pickupable.ItemSlot.Player, false);
+		}
+
+		private enum Mode
+		{
+			Off,
+			Wires,
+			Pipes,
+			Disposals
+		}
+
+		public string Examine(Vector3 worldPos = default(Vector3))
+		{
+			if (currentMode == Mode.Off)
+			{
+				return $"The {gameObject.ExpensiveName()} is off!";
+			}
+
+			return $"The {gameObject.ExpensiveName()} is currently detecting {currentMode.ToString()}";
+		}
+
+		#region Suicide
+
+		public bool CanSuicide(GameObject performer)
+		{
+			return true;
+		}
+
+		public IEnumerator OnSuicide(GameObject performer)
+		{
+			if (performer.TryGetComponent<LivingHealthMasterBase>(out var player) == false) yield break;
+
+			Chat.AddActionMsgToChat(performer, $"You begins to emit terahertz-rays into your brain with the {gameObject.ExpensiveName()}!",
```

Comment:
```suggestion
			Chat.AddActionMsgToChat(performer, $"You begin to emit terahertz-rays into your brain with the {gameObject.ExpensiveName()}!",
```

:eyes: Maybe add a warning to the description `A label reads, "do not point at living tissue"` : P